api/docker.ts

```diff
@@ -97,8 +121,217 @@ async function sizeHandler ({ scope, name, tag, architecture, variant }: PathArg
   const sizeInMegabytes = (imageData.size / 1024 / 1024).toFixed(2)
 
   return {
-    subject: 'docker image size',
+    subject: 'docker size',
     status: `${sizeInMegabytes} MB`,
     color: 'blue'
   }
 }
+
+async function layersHandler ({ scope, name, tag, architecture, variant }: PathArgs) {
+  tag = tag ? tag : 'latest'
+  architecture = architecture ? architecture : 'amd64'
+  variant = variant ? variant : ''
+
+  const token = (await getDockerAuthToken(scope, name)).token
+  
+  if (! token) {
+    return {
+      subject: 'docker metadata',
+      status: `unknown image: ${scope}/${name}`,
+      color: 'grey'
+    }
+  }
+
+  let headers = {
+    authorization: `Bearer ${token}`,
+    accept: `application/vnd.docker.distribution.manifest.list.v2+json`
+  }
+  let path = `v2/${scope}/${name}/manifests/${tag}`
+  
+  const manifest_list = (await queryDockerRegistry(path, headers)).manifests
+  
+  if (! manifest_list) {
+    return {
+      subject: 'docker layers',
+      status: `unknown tag: ${tag}`,
+      color: 'grey'
+    }
+  }
+
+  let manifest = manifest_list.find(manifest_list => manifest_list.platform.architecture === architecture)
+
+  if (! manifest) {
+    return {
+      subject: 'docker layers',
+      status: `unknown architecture: ${architecture}`,
+      color: 'grey'
+    }
+  }
+
+  if (variant) {
+    manifest = manifest_list.filter(manifest_list => manifest_list.platform.architecture === architecture).find(manifest_list => manifest_list.platform.variant === variant)
+
+    if (! manifest) {
+      return {
+        subject: 'docker layers',
+        status: `unknown variant: ${variant}`,
+        color: 'grey'
+      }
+    }
+  }
+
+  if (! manifest.digest) {
+    return {
+      subject: 'docker layers',
+      status: 'error getting image digest',
+      color: 'grey'
+    }
+  }
+
+  path = `v2/${scope}/${name}/manifests/${manifest.digest}`
+  const image_manifest = await queryDockerRegistry(path, headers)
+
+  if (! image_manifest) {
+    return {
+      subject: 'docker layers',
+      status: 'error getting image manifest',
+      color: 'grey'
+    }
+  }
+
+  const size = image_manifest.layers.map(layer => layer.size).reduce((accumulator, current) => accumulator + current, 0)
+  const sizeInMegabytes = (size / 1024 / 1024).toFixed(2)
+
+  headers = {
+    authorization: `Bearer ${token}`,
+    accept: `application/vnd.docker.image.config+json`
+  }
+  path = `v2/${scope}/${name}/blobs/${image_manifest.config.digest}`
+  const image_config = await queryDockerRegistry(path, headers)
+
+  if (! image_config) {
+    return {
+      subject: 'docker layers',
+      status: 'error getting image config',
+      color: 'grey'
+    }
+  }
+
+  const layers = image_config.history
+  if (! layers) {
+    return {
+      subject: 'docker layers',
+      status: 'error getting layers',
+      color: 'grey'
+    }
+  }
+
+  return {
+    subject: `${sizeInMegabytes} MB`,
```

Comment:
Let's keep the convention that subject/label is a static descriptive text

Reply:
This was done in commit [254fef7](https://github.com/badgen/badgen.net/pull/442/commits/254fef7691ab207ae87c188d1406d8082534b1a6)